### ISSUE TITLE
Registering Functions from Blueprints

### ISSUE THREAD

**Issue by lschumm** [jmcarp/flask-apispec#156]
Forgive me if I'm not understanding something, I'm not too familiar with Flask–

I'm working on a large Flask API that I'd like to document using `flask-apispec`. To organize the different sections of the API, we use Flask's Blueprint mechanism and pull all of these into a single `app` in a main file.

It seems that in your documentation you need to call `docs.register(view_function)` on every `view_function` defined in your app; which in our case would require a disastrous amount of imports and would be really painful to maintain. I came up with the following bit of code:

```python
for (fpath, view_function) in app.view_functions.items():
    blueprint_name = fpath.split('.')[0]
    # only bound methods have a .__self__ attribute, so we can check whether
    # a view_function is a defined route or something included from a package or from Flask
    # by using hasattr.
    if not hasattr(view_function, '__self__'):
        docs.register(view_function, blueprint=blueprint_name)
```

Is this good practice, or is there a better way to do this within `flask-apispec`?

**Comment by SmoothMaverick** [comment]
I have the same issue.

**Comment by paxnovem** [comment]
Currently as far as I am aware the current two methods for registering is by view_functions or by view_classes. If I am understanding things correctly, supporting blueprint as a target for the register method would resolve your usecase. Is that correct?

_Note_
It would probably involve adding blueprint as an accepted type and add a BlueprintConverter to the apidoc file. 

In regards to your workaround for this, I am not familiar enough to know whether the assumption that view_functions are always the only unbound methods returned in all cases. That being said if it were true, I would probably use the native inspect method. https://docs.python.org/3/library/inspect.html#inspect.ismethod rather than explicitly checking for __self__.

**Comment by lschumm** [comment]
@paxnovem Yeah, my "solution" is extremely hacky; good point about the `__self__` stuff. Since `flask-apispec` didn't really meet the needs of the API I've been working on, I implemented a sort of mini-library that does similar. My approach was that the `@doc` decorator sets the `__apispec__` attribute of each endpoint function to the contents passed into `doc`. When iterating over all endpoints in the application to generate the documentation, each is now tagged and coupled with its documentation to make aggregation super easy.
